validate_test.go

```diff
@@ -0,0 +1,97 @@
+package gitlab
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"testing"
+)
+
+// XXX: move this func out if necessary
+// content read a file, return its content as string
+func content(f string) string {
+	b, err := ioutil.ReadFile(f)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
`log.Fatal(err)` does the same as `fmt.Println(err)` and `os.Exit(1)` do together. But maybe just return an error here as well and use `t.Fatal(err)` in the calling function instead?

Reply:
Suggest trying to avoid `os.Exit` (and `log.Fatal`) in tests. The situation might be better in go1.10, but generally speaking this breaks generating coverage when testing in these cases. If its purely in a test context though as @svanharmelen mentions `t.Fatal` should be safe. having this return an error seems solid

Reply:
Might even want to just inline this function in the test after changing the tests to be table driven